docs/specs/#1984 - Apps and Packages Spec/spec.md

```diff
@@ -0,0 +1,118 @@
+---
+author: Sharla Soennichsen @shakersMSFT and Jordi Adoumie @joadoumie
+created on: 2024-01-30
+last updated: 2024-01-30
+issue id: 1984
+---
+
+# Apps & Packages
+
+## 1. Overview
+
+### 1.1 Establish the Problem
+
+Devs often need to install and manage a variety of tools and packages to get their work done. This can be a time consuming and frustrating process scattered across various UIs and CLIs.
+
+### 1.2 Introduce the Solution
+
+This new feature aims to create an all-in-one apps & packages management experience in Dev Home that allows you to manage your apps & package in a single, streamlined interface. From a unified interface, users can seamlessly handle package installation (exists in Dev Home already today), updates, and rollbacks for their winget packages. These enhancements will significantly boost efficiency, convenience, and control for software developers and power users, transforming the way they manage apps & packages.
+
+### 1.3 Rough-in Designs
+
+![Apps and Packages View 1](apps-and-packages.png)
+
+![Update and View](update-and-view.png)
+
+## 2. Goals & User Cans
+
+### 2.1 Goals
+
+| No. | Goal | Priority |
+| --- | ---- | -------- |
+| 1   |   Provide users with an easy and delightful experience to manage installed applications.    |     0     |
+| 2   |   Incorporate advanced package management features such as version management, batch operations, automatic updates, and rollback capabilities.    |     0     |
+
+### 2.2 Non-Goals
+
+| No. | Non-Goal |
+| --- | -------- |
+| 1   |     This feature is not intended to replace the Windows Settings 'Apps > Installed' apps functionality.     |
+
+### 2.3 User Cans Summary Table
+
+| No. | User Can | Priority |
+| --- | -------- | -------- |
+| 1   |     A user can view application name, publisher, size, version, and source      |     0     |
+| 2   |    A user can switch between different versions of a package       |    0      |
+| 3   |    A user can select multiple applications at a time to perform bulk actions like update or uninstall       |    0      |
+| 4   |   A user can search to find an application       |      0    |
+| 5   |     A user can filter the page by update available      |      0    |
+| 6   |    A user can sort the page by application name, publisher, size, and source.      |    0      |
+
+## 3. User Stories
+
+### 3.1 User story - Updating and Rolling Back Packages
+
+A software developer wants to update a package to the latest version but finds that it has a bug affecting their system.
+
+#### Job-to-be-done
+
+1. Update a package to the latest version
+2. Roll back to a previous version of a package
```

Comment:
How do you want dependencies to be handled when rolling back?  If Package Bv3.0 depends on Package Av2.0.  Package A is rolled back to v1.0.  What should happen to package B?  Should it roll back?  Should the roll back terminate with an unsuccessful message because other packages need to be rolled back?